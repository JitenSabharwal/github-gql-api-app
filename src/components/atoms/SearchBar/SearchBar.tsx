import React, { useState } from "react"
import styled from "styled-components"
import {
  InputGroup,
  InputGroupButtonDropdown,
  Input,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from "reactstrap"
const Root = styled.div`
  display: flex;
`
enum IssueState {
  Open = "Open",
  Closed = "Closed",
}
export default function SearchBar() {
  const [dropdownOpen, setDropdownOpen] = useState(false)
  const [issueState, setIssueState] = useState(IssueState.Open)
  const toggleDropDown = () => setDropdownOpen(!dropdownOpen)
  const updateIssueState = (state: IssueState) => setIssueState(state)
  return (
    <Root>
      <InputGroup>
        <InputGroupButtonDropdown
          addonType="append"
          isOpen={dropdownOpen}
          toggle={toggleDropDown}
        >
          <DropdownToggle caret>Issue: {issueState}</DropdownToggle>
          <DropdownMenu>
            <DropdownItem header>Issue Status</DropdownItem>
            <DropdownItem onClick={() => updateIssueState(IssueState.Open)}>
              Open
            </DropdownItem>
            <DropdownItem divider />
            <DropdownItem onClick={() => updateIssueState(IssueState.Closed)}>
              Closed
            </DropdownItem>
          </DropdownMenu>
        </InputGroupButtonDropdown>
        <Input onChange={() => {}} type="text" />
      </InputGroup>
    </Root>
  )
}
