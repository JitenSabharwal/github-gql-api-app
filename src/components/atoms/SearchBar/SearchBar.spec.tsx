import React from "react"
import SearchBar from "./SearchBar"
import { shallow } from "enzyme"

describe("SearchBar", () => {
  let container: React.ReactElement
  beforeAll(() => (container = <SearchBar />))
  it("Should render SearchBar", () => {
    const component = shallow(container)
  })
  it("Should render the initial layout", () => {
    const component = shallow(container)
    expect(component.getElements()).toMatchSnapshot()
  })
})
