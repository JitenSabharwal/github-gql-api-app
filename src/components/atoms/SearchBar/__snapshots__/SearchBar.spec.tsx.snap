// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should render the initial layout 1`] = `
Array [
  <styled.div>
    <InputGroup
      tag="div"
    >
      <InputGroupButtonDropdown
        addonType="append"
        isOpen={false}
        toggle={[Function]}
      >
        <DropdownToggle
          aria-haspopup={true}
          caret={true}
          color="secondary"
        >
          Issue: 
          Open
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          tag="div"
        >
          <DropdownItem
            header={true}
            tag="button"
            toggle={true}
          >
            Issue Status
          </DropdownItem>
          <DropdownItem
            onClick={[Function]}
            tag="button"
            toggle={true}
          >
            Open
          </DropdownItem>
          <DropdownItem
            divider={true}
            tag="button"
            toggle={true}
          />
          <DropdownItem
            onClick={[Function]}
            tag="button"
            toggle={true}
          >
            Closed
          </DropdownItem>
        </DropdownMenu>
      </InputGroupButtonDropdown>
      <Input
        onChange={[Function]}
        type="text"
      />
    </InputGroup>
  </styled.div>,
]
`;
